crates/holochain/tests/tests/regression/two_apps_same_dna_hash_different_coordinators.rs

```diff
@@ -0,0 +1,117 @@
+use hdk::prelude::{CoordinatorZome, IntegrityZome, Record, SerializedBytes};
+use holo_hash::ActionHash;
+use holochain::{
+    prelude::DnaWasm,
+    sweettest::{SweetConductor, SweetDnaFile},
+};
+use holochain_wasm_test_utils::{TestCoordinatorWasm, TestIntegrityWasm};
+
+/// Regression test for https://github.com/holochain/holochain/issues/2145#issuecomment-2460304757
+#[tokio::test(flavor = "multi_thread")]
+async fn can_install_two_apps_with_the_same_dna_hash_but_different_coordinators() {
+    let dna_file_1 = SweetDnaFile::from_zomes(
+        "the same seed".into(),
+        vec![IntegrityZome::from(TestIntegrityWasm::IntegrityZome)],
+        vec![CoordinatorZome::from(TestCoordinatorWasm::CoordinatorZome)],
+        vec![
+            DnaWasm::from(TestIntegrityWasm::IntegrityZome),
+            DnaWasm::from(TestCoordinatorWasm::CoordinatorZome),
+        ],
+        SerializedBytes::default(),
+    )
+    .await
+    .0;
+
+    // Create another DnaFile with a different coordinator zome
+    let dna_file_2 = SweetDnaFile::from_zomes(
+        "the same seed".into(),
+        vec![IntegrityZome::from(TestIntegrityWasm::IntegrityZome)],
+        vec![CoordinatorZome::from(
+            TestCoordinatorWasm::CoordinatorZomeUpdate,
+        )],
+        vec![
+            DnaWasm::from(TestIntegrityWasm::IntegrityZome),
+            DnaWasm::from(TestCoordinatorWasm::CoordinatorZomeUpdate),
+        ],
+        SerializedBytes::default(),
+    )
+    .await
+    .0;
+
+    // Verify that the dna hashes match
+    assert_eq!(dna_file_1.dna_hash(), dna_file_2.dna_hash());
+
+    // Install the first app and write an entry to its integrity zome
```

Comment:
Entries are not written to integrity zomes, so I suggest a different wording. I understand that you mean that both apps use a DNA with identical integrity zomes and network seeds and therefore are on the same network. The entries are written to different source chains and then copied over to a DHT that is shared per DNA. That's why the entry written in app 1 can be read by app 2.
```suggestion
    // Install the first app and write an entry
```

Reply:
Agreed. I copied that wording over from the update_coordinators test I believe. I will update it there as well.

Reply:
https://github.com/holochain/holochain/pull/5189/commits/9fb668a02800c85d7a68f4d94c2d4baf27fe9d2c